clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -451,21 +456,33 @@ raw2trace_t::merge_and_process_thread_files()
                tidx, (int)thread_files[tidx]->tellg());
         if (!thread_files[tidx]->read((char*)&in_entry, sizeof(in_entry))) {
             if (thread_files[tidx]->eof()) {
-                VPRINT(2, "Finish read file %d\n", tidx);
-                times[tidx] = 0;  // Do not read from this file.
-                tidx = (uint)thread_files.size(); // Request thread scan.
-                --file_count;
-                continue;
+                // Rather than a FATAL_ERROR we try to continue to provide partial
+                // results in case the disk was full or there was some other issue.
+                WARN("Input file %d is truncated", tidx);
+                in_entry.extended.type = OFFLINE_TYPE_EXTENDED;
+                in_entry.extended.ext = OFFLINE_EXT_TYPE_FOOTER;
             } else
                 FATAL_ERROR("Failed to read from file %d", tidx);
         }
         if (in_entry.extended.type == OFFLINE_TYPE_EXTENDED) {
             if (in_entry.extended.ext == OFFLINE_EXT_TYPE_FOOTER) {
+                VPRINT(2, "Finish reading file %d at pos %u\n",
+                       tidx, (int)thread_files[tidx]->tellg());
+                // We need do another failed read to make ->eof() return true.
+                if (thread_files[tidx]->read((char*)&in_entry, 1)) {
```

Comment:
The previous code that you deleted seems better, "if (read entry || !eof) fatal..."